tensorflow/core/kernels/data/batch_dataset_op_test.h

```diff
@@ -0,0 +1,86 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_CORE_KERNELS_DATA_BATCH_DATASET_OP_TEST_H
+#define TENSORFLOW_CORE_KERNELS_DATA_BATCH_DATASET_OP_TEST_H
+
+#include "tensorflow/core/kernels/data/batch_dataset_op.h"
+#include "tensorflow/core/kernels/data/dataset_test_params.h"
+
+namespace tensorflow {
+namespace data {
+
+class BatchDatasetParams : public DatasetParams {
```

Comment:
I'm in favor of having a dataset test utils file / namespace where these common classes are defined, instead of having a header file for batch dataset test specifically. It makes the test harder to read. Furthermore, you already have a bunch of utility-like files `dataset_test_{base | params}.{h | cc}`. I'd recommend coalescing all that into a single pair of `.h` and `.cc` files.

 I'll defer to @jsimsa on this though, if he thinks otherwise.

Reply:
There are two kinds of `DatasetParams`: 
1) the first one (e.g. `RangeDatasetParams`) will be used as the input dataset by other dataset op tests; 
2) the second kind of DatasetParams (e.g. `ReduceDatasetParams`) will be only used by itself. 

For the first one, we need to define it in some files (e.g. `dataset_test_params.h`) that can be imported by other tests; for the second one, we can define the dataset params class in its own test file (e.g. `reduce_dataset_op_test.cc`) as it will not be used by other tests. It may solve the scale issue that @jsimsa mentioned before.

`dataset_test_base.h` already has more than 600 lines of code, so I did not combine `dataset_test_base.{h | cc}` and `dataset_test_params.{h | cc}` together. If the length of the file is ok,  I will merge them into a single pair of `.h` and `.cc` file once @jsimsa and @rachellim make the final decision. 

Adding a namespace for the tests (e.g. `tensorflow::data::testing`) sounds great to me.

Reply:
@feihugis if we only expect a small number of dataset parameter definitions to be shared, then @rachellim's suggestion makes sense.

I would not worry about the size of the `dataset_test_base` files as long as they are well organized and documented.

In summary, I support:
1) coalescing all test related functionality in `dataset_test_base.{h, cc}` putting shared dataset parameter definitions there and inlining all dataset parametr definitions in the `*_test.cc` files.

Reply:
Great, will coalesce these `_params.h` files into  the `dataset_test_base` files.